tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -449,6 +449,15 @@ spec:
 			stdOut = helper.CmdShouldFail("odo", "link", "EtcdCluster/example")
 			Expect(stdOut).To(ContainSubstring("already linked with the service"))
 
+			// Before running "odo unlink" checks, wait for the pod to come up from "odo push" done after "odo link"
+			pods = oc.GetAllPodsInNs(project)
```

Comment:
Listing all pods in not a namespace is not nice.
If you need pods for the given component you can do the equivalent of `kubectl get deployment --selector component=<componentname>` 

It looks like we already have a function that might be  doing what you are trying to do 
https://github.com/openshift/odo/blob/2180507a4850dfac98559cc94aa9d55441007c0b/tests/helper/helper_kubectl.go#L69

Reply:
this function might be also usefull for you https://github.com/openshift/odo/blob/2180507a4850dfac98559cc94aa9d55441007c0b/tests/helper/helper_kubectl.go#L110

Reply:
> GetRunningPodNameByComponent

This one looks useful. It does help me get the exact pod name that was finding through regex.

> WaitAndCheckForExistence

Looking at this function's comment and implementation, I think it's going to check if the resource type has ceased to exist. Whereas what I'm looking for is the pod to be in `Running` state.

I am going to use `GetRunningPodNameByComponent` and run it a few times against CRC before pushing a commit.

Reply:
I changed the code to use `GetRunningPodNameByComponent` method. Although I don't think it has anything to do with changing the way we wait for pods to come up, the test failed once during the 10 runs I tried it out for. With the `GetAllPodsInNS` implementation, it didn't fail. I think it's unrelated and probably indicates that we're still missing something to fix this particular flake.

At this point, I can't think of anything else and would need someone else taking a look at things if #4301 becomes an annoyance again.

Reply:
I'm going to roll back to using `oc.GetAllPodsInNs(project)`. I'm doing this because:
1. https://github.com/openshift/odo/pull/4428#issuecomment-782079225
2. I am going to need some help if this failure becomes a severe annoyance again. I can't really find what's going wrong. Very likely I'm missing something.